xarray/core/formatting.py

```diff
@@ -142,7 +142,7 @@ def format_item(x, timedelta_format=None, quote_strings=True):
         return format_timedelta(x, timedelta_format=timedelta_format)
     elif isinstance(x, (str, bytes)):
         return repr(x) if quote_strings else x
-    elif np.issubdtype(type(x), np.floating):
+    elif hasattr(x, 'dtype') and np.issubdtype(x.dtype, np.floating):
```

Comment:
This looks very reasonable. Is the case where `hasattr(x, 'dtype')` is False tested?

Reply:
I've made this change because 0D numpy arrays (which is what you get from da.coords['e'].values when 'e' is a scalar numeric coord) have a `numpy.ndarray` type, which is not a subtype of `np.floating`. If `hasattr(x, 'dtype')` is false then it falls back to the `else` clause and the object string representation. Do you have another case in mind not covered by these checks?

Reply:
Great, thanks for explaining so clearly